tools/tool.py

```diff
@@ -0,0 +1,355 @@
+        return Status(), self.ParseResult(output)
+
+    def CancelOp(self, database, op_id) -> Status:
+        cmd = list(self.ns_base_cmd)
+        cmd.append("--cmd=cancelop {}".format(op_id))
+        cmd.append("--database=" + database)
+        status, output = self.RunWithRetuncode(cmd)
+        return status
+
+    def Migrate(self, database, name, pid, src_endpoint, desc_endpoint, sync = False) -> Status:
+        if src_endpoint == desc_endpoint:
+            return Status(-1, "src_endpoint and desc_endpoint is same")
+        cmd = list(self.ns_base_cmd)
+        cmd.append("--cmd=migrate {} {} {} {}".format(src_endpoint, name, pid, desc_endpoint))
+        cmd.append("--database=" + database)
+        status, output = self.RunWithRetuncode(cmd)
+        if status.OK() and output.find("migrate ok") != -1:
+            if sync and not self.WaitingOP(database, name, pid).OK():
+                return Status(-1, "migrate failed")
+            return Status()
+        return status
+
+    def ShowTablet(self) -> tuple([Status, List]):
+        cmd = list(self.ns_base_cmd)
+        cmd.append("--cmd=showtablet")
+        status, output = self.RunWithRetuncode(cmd)
+        if not status.OK():
+            return status, None
+        return Status(), self.ParseResult(output)
+
+    def AddReplica(self, database, name, pid, endpoint, sync = False) -> Status:
+        cmd = list(self.ns_base_cmd)
+        cmd.append("--cmd=addreplica {} {} {}".format(name, pid, endpoint))
+        cmd.append("--database=" + database)
+        status, output = self.RunWithRetuncode(cmd)
+        if status.OK() and output.find("ok") != -1:
+            if sync and not self.WaitingOP(database, name, pid).OK():
+                return Status(-1, "addreplica failed")
+            return Status()
+        return Status(-1, "add replica failed")
+
+    def DelReplica(self, database, name, pid, endpoint, sync = False) -> Status:
+        cmd = list(self.ns_base_cmd)
+        cmd.append("--cmd=delreplica {} {} {}".format(name, pid, endpoint))
+        cmd.append("--database=" + database)
+        status, output = self.RunWithRetuncode(cmd)
+        if status.OK() and output.find("ok") != -1:
+            if sync and not self.WaitingOP(database, name, pid).OK():
+                return Status(-1, "delreplica failed")
+            return Status()
+        return Status(-1, "del replica failed")
+
+    def WaitingOP(self, database, name, pid) -> Status:
+        while True:
+            error_try_times = 3
+            while error_try_times > 0:
+                status, result = self.ShowOpStatus(database, name, pid)
+                error_try_times -= 1
+                if status.OK():
+                    break
+                elif error_try_times == 0:
+                    return Status(-1, "fail to execute showopstatus")
+            is_finish = True
+            for record in result:
+                if record[4] == 'kDoing' or record[4] == 'kInited':
```

Comment:
if `kFailed`, should return `non-ok` status

Reply:
there may have history failed jobs.